cnd/src/http_api.rs

```diff
@@ -143,41 +143,32 @@ impl Amount {
 
 #[derive(Serialize)]
 struct State {
-    #[serde(serialize_with = "percent_string")]
-    open: u8,
-    #[serde(serialize_with = "percent_string")]
-    closed: u8,
-    #[serde(serialize_with = "percent_string")]
-    settling: u8,
-    #[serde(serialize_with = "percent_string")]
-    failed: u8,
-    #[serde(serialize_with = "percent_string")]
-    cancelled: u8,
+    open: Amount,
+    closed: Amount,
+    settling: Amount,
+    failed: Amount,
+    cancelled: Amount,
 }
 
 impl State {
     #[allow(clippy::cast_sign_loss, clippy::cast_possible_truncation)] // we only store positive values in the DB ranging from 0 - 100
-    fn new(open: i32, closed: i32, settling: i32, failed: i32, cancelled: i32) -> Self {
+    fn new(
+        open: Amount,
+        closed: Amount,
+        settling: Amount,
+        failed: Amount,
+        cancelled: Amount,
+    ) -> Self {
         Self {
```

Comment:
`State::new` is only called once in `http_api`. You don't need it, you can directly construct the State using the fields. Like it's done here: https://github.com/comit-network/comit-rs/blob/9d3157a0c088266c70057f0c2c7bc30ceb5d1772/cnd/src/http_api.rs#L519